app/models/pipeline_run.rb

```diff
@@ -153,38 +160,249 @@ def retry
   end
 
   def report_ready?
-    job_status == STATUS_CHECKED || (ready_step && active_stage && active_stage.step_number > ready_step)
+    results_finalized == 1 || (ready_step && active_stage && active_stage.step_number > ready_step)
   end
 
   def succeeded?
     job_status == STATUS_CHECKED
   end
 
-  def update_job_status
-    prs = active_stage
-    all_stages_succeeded = prs.nil?
-    if all_stages_succeeded
-      self.finalized = 1
-      self.job_status = STATUS_CHECKED
+  def db_load_host_filtering
+    # Load version info applicable to host filtering
+    version_s3_path = "#{host_filter_output_s3_path}/#{VERSION_JSON_NAME}"
+    self.version = `aws s3 cp #{version_s3_path} -`
```

Comment:
I know this was there before, but this is a little weird isn't it? is it supposed to be self.version == 0 meaning the version file was copied successfully?

Reply:
Hm? It's actually a JSON that contains info about the versions of the host genomes used during filtering etc

Reply:
What this command does is copy the file from S3 and insert the contents directly into self.version without the need for a temp file :)